gpytorch/lazy/toeplitz_lazy_variable.py

```diff
@@ -155,18 +156,18 @@ def mul(self, other):
         """
         if isinstance(other, ToeplitzLazyVariable):
             if len(self.c) != len(other.c):
-                raise RuntimeError('Can only multiply two ToeplitzLazyVariables if they correspond to the same grid.')
+                return MulLazyVariable(self, other)
             if self.J_left is not None:
                 if other.J_left is None:
-                    raise RuntimeError('Cannot multiply interpolated ToeplitzLazyVariables by non-interpolated ones.')
+                    return MulLazyVariable(self, other)
                 if not (approx_equal(self.C_left, other.C_left) and approx_equal(self.C_right, other.C_right)):
-                    raise RuntimeError('Cannot multiply two ToeplitzLazyVariables with different',
-                                       'left interpolation matrices.')
```

Comment:
I think with `MulLazyVariable`, the check we want is simply if the sizes are the same, since that's the only requirement for elementwise multiplication to work. It should be on the user's head if they elementwise two `ToeplitzLazyVariables` they didn't intend to. In other words, if `other` is a `ToeplitzLazyVariable` of the same size we can return a `MulLazyVariable`.

Reply:
For now we can leave it throwing an error if they both have added diagonal components -- the logic of FOILing (A+p*I)*(B+q*I) will be much simpler once we implement added diagonal components as `DiagLazyVariable`